packages/playwright-core/src/server/browserType.ts

```diff
@@ -77,11 +77,20 @@ export abstract class BrowserType extends SdkObject {
   async launchPersistentContext(metadata: CallMetadata, userDataDir: string, options: channels.BrowserTypeLaunchPersistentContextOptions & { useWebSocket?: boolean }): Promise<BrowserContext> {
     options = this._validateLaunchOptions(options);
     const controller = new ProgressController(metadata, this);
-    const persistent: channels.BrowserNewContextParams = options;
+    const persistent: channels.BrowserNewContextParams = { ...options };
     controller.setLogName('browser');
+
+    // Note: Any initial TLS requests will fail since we rely on the Page/Frames initialize which sets ignoreHTTPSErrors.
```

Comment:
I'd move lines 83-86 and 92 inside the `controller.run()` to be governed by `progress` and have a timeout.